tests/contrib/BUILD

```diff
@@ -21,6 +21,7 @@ load("//contrib:compare_ids_test.bzl", "compare_ids_test")
 load("//contrib:idd.bzl", "idd")
```

Comment:
Add a test that uses this new test rule in an external repository (something similar to https://github.com/bazelbuild/rules_docker/tree/master/testing/java_image, could be in https://github.com/bazelbuild/rules_docker/tree/master/testing/repro_tests/basic)

Also add a test (in a different external repo, https://github.com/bazelbuild/rules_docker/tree/master/testing/repro_tests/toolchain_container) that uses a toolchain_container target that includes packages and gpg_keys.

Also add a test that fails, for a toolchain_container that includes a package that we know is not deterministic unless we do some additional processing of it e.g., https://github.com/GoogleCloudPlatform/layer-definitions/blob/master/layers/ubuntu1604/python/BUILD#L52

For all these three tests set up GCB to run them and verify their success/failure (for the last one). You'll need to also enable docker sibling on the GCB run, probably 'just as easy' as passing `-v /var/run/docker.sock:/var/run/docker.sock`

Reply:
Added simple tests under tests/contrib for now.
Adding tests in external repos to run in GCB is more complex and will be attempted in a different PR.